File: ArticleProject.Models/ArticleModels/Article.cs
﻿using ArticleProject.Models.CategoryModels;
using System.Reflection.Metadata;

namespace ArticleProject.Models.ArticleModels
{
    public class Article
    {
        public string Id { get; set; }

        public string Title { get; set; }

        public string Description { get; set; }

        public string Photo { get; set; }

        public string CategoryName { get; set; }

        public string UserId { get; set; }

        public Comment[] Comments { get; set; }
    }
}

File: ArticlesProjectNg/src/app/articles-management/models/article.ts
import { UserComment } from "./user-comment";
import { UserBrief } from "app/registration-management/models/userBrief";

export class Article {
    constructor(
        public id: string,
        public title: string,
        public photo: any,
        public createdDate: string,
        public description: string,
        public user: UserBrief,
        public categoryName: string,
        public comments: UserComment[],
    ) { }
}